### PR TITLE
BM-396: initial work to build middleware

### PR DESCRIPTION
**Pull request by JoeChapman**
- add all project components
- travis config
- dotfiles
  ...

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -0,0 +1,5 @@
+language: node_js
+
+node_js:
+  - "4"
```
Comment (chrisns) [inline .travis.yml]: probably better to test against `4.2.2` which the LTS release and `stable`

File: index.js [hunk 2]
```diff
@@ -0,0 +1,52 @@
+'use strict';
+
+var transforms = require('./transforms');
+
+var applyTransforms = (item, req, prop) => {
+  item.transforms = [].concat(item.transforms);
```
Comment (chrisns) [inline index.js:6]: is there variable reassignment here?
if so is there a better way to express what you want?

File: index.js [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+
+module.exports = (req, res, next) => {
+  res.body = {customfields: {}};
+
+  Object.defineProperty(res, 'customfields', {
+    get: () => {
+      return res.body.customfields;
+    },
+    set: (items) => {
+      items = [].concat(items);
```
Comment (chrisns) [inline index.js:44]: variable reassignment

File: index.js (hunk 2 above)
Comment (JoeChapman) [inline index.js:6]: Are you also bothered by the variable reassignment to `prop` on line 9, 11, and 13 in `index.js`?

**Comment by chrisns** [comment]
it will if you put one there and rely on maintaining that, this just sets the version number to 1.0.$travis-build-number and ignores whatever is in the package.json
The better solution would be to move to tagged releases and make npm deploy when you tag a version.

**Pull request merged**